src/main/java/Duke.java

```diff
@@ -1,10 +1,136 @@
+
+        //when the program starts
+        showLogo();
+        greet();
+
+        while(!isQuit){
+            processInput(in);
+
+            switch (command.toUpperCase()) {
+            case "LIST":
+                listTasks();
+                break;
+            case "DONE":
+                doneTask(Integer.parseInt(description));
+                break;
+            case "TODO":
+                addTask(new ToDo(description));
+                break;
+            case "DEADLINE":
+                processDeadline(description);
+                addTask(new Deadline(taskName,taskTime));
+                break;
+            case "EVENT":
+                processEvent(description);
+                addTask(new Event(taskName,taskTime));
+                break;
+            case "BYE":
+                isQuit = true;
+                bye();
+                break;
+            case "HELP":
+                printHelpInfo();
+                break;
+            default:
+                printErrorInfo();
+                break;
+            }
+        }
+    }
+
+    private static void processInput(Scanner in) {
+        userInputLine = in.nextLine();
+        command = processCommand();
+        description = processDescription(command);
+    }
+
+    private static void printErrorInfo() {
+        System.out.println("The input is invalid, pls refer to the commands by typing 'help'. ");
+    }
+    private static void printHelpInfo() {
+        System.out.println(HORIZONTAL_LINE + "\n"
+                + "LIST, DONE, TODO, DEADLINE, EVENT, BYE, HELP"
+                + HORIZONTAL_LINE);
+    }
+
+    public static void showLogo(){
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        System.out.println(HORIZONTAL_LINE+ "\n" + logo);
+    }
+    public static void greet(){
+        String greeting = "Hello! I'm Duke\n"
+                        + "What can I do for you?\n";
+        System.out.println(HORIZONTAL_LINE + "\n" + greeting + HORIZONTAL_LINE);
+    }
+    public static void bye(){
+        String bye_word = "Bye. Hope to see you again soon!";
+        System.out.println(HORIZONTAL_LINE + "\n" + bye_word + "\n" + HORIZONTAL_LINE);
+    }
+
+    /*public static void echoCommand(String userInput){
+        System.out.println(HORIZONTAL_LINE + "\n" + userInput + "\n" + HORIZONTAL_LINE);
+    }*/
+
+    public static void addTask(Task t){
+        tasks[taskCount] = t;
+        taskCount ++;
+        System.out.println(HORIZONTAL_LINE + "\n"
+                + "Got it. I've added this task: \n"
+                + "  " +t.toString() + "\n"
+                + "Now you have "+ taskCount + " tasks in the list \n"
+                + HORIZONTAL_LINE);
+    }
+    public static void listTasks(){
```

Comment:
same issue here, minor indentation error

Reply:
hi Siew wen Thanks for the review! I should put a space between my close bracket and open brace.